### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +15,143 @@ const goodsFromServer: string[] = [
...
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  listIsVisible: boolean,
+  copyOfGoods: string[],
+  initialGoods: string[],
+  listIsReversed: boolean,
+  sortby: string,
```
Comment (dellachaise) [inline src/App.tsx:23]: Please change to sortBy, because we use camelCase for naming here.

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,143 @@ const goodsFromServer: string[] = [
...
+        listIsReversed: !prevstate.listIsReversed,
+      }
+    ));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(prevstate => (
+      {
+        copyOfGoods: [...prevstate.initialGoods],
+        sortby: 'alphabet',
```
Comment (dellachaise) [inline src/App.tsx:54]: Please change to sortBy, because we use camelCase for naming here.

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +15,143 @@ const goodsFromServer: string[] = [
...
+      }
+    ));
+  };
+
+  sortByLength = () => {
+    this.setState(prevstate => (
+      {
+        copyOfGoods: [...prevstate.initialGoods],
+        sortby: 'length',
```
Comment (dellachaise) [inline src/App.tsx:63]: Please change to sortBy, because we use camelCase for naming here.

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +15,143 @@ const goodsFromServer: string[] = [
...
+      }
+    ));
+  };
+
+  render() {
+    const {
+      listIsReversed, listIsVisible, copyOfGoods, sortby,
+    } = this.state;
+
+    if (sortby === 'length') {
```
Comment (dellachaise) [inline src/App.tsx:83]: Please change to sortBy, because we use camelCase for naming here.

File: src/App.tsx [hunk 5]
```diff
@@ -14,11 +15,143 @@ const goodsFromServer: string[] = [
...
+      copyOfGoods.sort((a, b) => {
+        return (a.length - b.length);
+      });
+    }
+
+    if (sortby === 'alphabet') {
```
Comment (dellachaise) [inline src/App.tsx:89]: Please change to sortBy, because we use camelCase for naming here.

**Review by dellachaise — CHANGES_REQUESTED** [review]
Good job!

But sample on DemoLink works improperly. Please check it.

File: src/App.tsx [hunk 6]
```diff
@@ -14,11 +15,145 @@ const goodsFromServer: string[] = [
...
+
+    if (sortBy === 'length') {
+      copyOfGoods.sort((a, b) => {
+        return (a.length - b.length);
+      });
+    }
+
+    if (sortBy === 'alphabet') {
+      copyOfGoods.sort();
+    }
```
Comment (Denchepornyuk) [inline src/App.tsx:93]: use localeCompare for sorting strings

File: src/App.tsx [hunk 7]
```diff
@@ -14,11 +15,145 @@ const goodsFromServer: string[] = [
...
+
+// eslint-disable-next-line react/prefer-stateless-function
+class App extends React.Component<{}, State> {
+  state = {
+    listIsVisible: false,
+    // eslint-disable-next-line react/no-unused-state
+    copyOfGoods: goodsFromServer,
+    listIsReversed: false,
+    initialGoods: goodsFromServer,
+    sortBy: '',
```
Comment (Denchepornyuk) [inline src/App.tsx:34]: do you need copyOfGoods and initialGoods?
I believe you can use only one array for goods in state

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**